pyomo/contrib/mindtpy/feasibility_pump_new.py

```diff
@@ -0,0 +1,205 @@
+
+# -*- coding: utf-8 -*-
+
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import logging
+from pyomo.contrib.mindtpy.config_options import _get_MindtPy_FP_config
+from pyomo.contrib.mindtpy.algorithm_base_class import _MindtPyAlgorithm
+from pyomo.opt import TerminationCondition as tc
+from pyomo.core import minimize, Constraint, TransformationFactory, value, Objective, ConstraintList
+from pyomo.contrib.mindtpy.feasibility_pump import generate_norm_constraint, fp_converged, add_orthogonality_cuts
```

Comment:
Why not declare these functions in this file and stick to one `feasibility_pump.py`? Or if you want to keep them separate, could you rename this file--it won't always be new...

Reply:
Good suggestion.